internal/dcache/rpc/server/handler.go

```diff
@@ -1878,6 +1879,241 @@ refreshFromClustermapAndRetry:
 	return resp, nil
 }
 
+// PutChunkEx RPC takes the PutChunkRequest and a list of next RVs to forward the request to as parameters.
+// The PutChunkRequest is for the local RV. If the RV mentioned in the PutChunkRequest is not the local RV,
+// it will return an InvalidRVID error.
+// Parallelly, it also forwards the PutChunkRequest to the next RVs in the list.
+//
+// For the local RV, it calls the PutChunk RPC via the handler directly.
+// The errors returned by the PutChunk calls to the local RVs and next RVs are returned in the
+// PutChunkExResponse, with the RV name as the key and its PutChunkResponse or error as the value.
+// The PutChunkExResponse will have the responses for all the RVs in the list, including the local RV.
+// If the PutChunkExRequest is for an RV that is not hosted by this node, it will return an InvalidRVID error.
+func (h *ChunkServiceHandler) PutChunkEx(ctx context.Context, req *models.PutChunkExRequest) (*models.PutChunkExResponse, error) {
+	// Thrift should not be calling us with nil req.
+	common.Assert(req != nil)
+
+	// Thrift should not be calling us with nil Request, Chunk or Address.
+	common.Assert(req.Request != nil)
+	common.Assert(req.Request.Chunk != nil)
+	common.Assert(req.Request.Chunk.Address != nil)
+
```

Comment:
also assert that req.NextRVs is not empty. Caller should call us PutChunkEx only if it needs to be fwded to at least one, else it should simply call PutChunk

Reply:
Added